.github/workflows/node-ci.yml

```diff
@@ -118,63 +123,98 @@ jobs:
             libc++abi-9-dev
           /usr/sbin/update-ccache-symlinks
 
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
+
       - name: npm ci
         run: npm ci --ignore-scripts
 
-      - name: Prepare ccache
-        run: ccache --clear --set-config cache_dir=~/.ccache
+      - name: Set up msvc dev cmd (Windows)
+        if: runner.os == 'Windows'
+        uses: ilammy/msvc-dev-cmd@v1
 
-      - name: Cache ccache
-        uses: actions/cache@v3
-        env:
-          cache-name: ccache-v1
-        with:
-          path: ~/.ccache
-          key: ${{ env.cache-name }}-${{ matrix.os }}-${{ github.job }}-${{ github.ref }}-${{ github.sha }}-${{ github.head_ref }}
-          restore-keys: |
-            ${{ env.cache-name }}-${{ matrix.os }}-${{ github.job }}-${{ github.ref }}-${{ github.sha }}
-            ${{ env.cache-name }}-${{ matrix.os }}-${{ github.job }}-${{ github.ref }}
-            ${{ env.cache-name }}-${{ matrix.os }}-${{ github.job }}
-
-      - name: Clear ccache statistics
+      - name: Update ccache (Windows)
+        if: runner.os == 'Windows'
         run: |
-          ccache --zero-stats --set-config cache_dir=~/.ccache
-          ccache --max-size=2G --set-config cache_dir=~/.ccache
-          ccache --show-stats --set-config cache_dir=~/.ccache
+          choco.exe upgrade ccache
+          ccache.exe --version
+          echo "CCACHE_CONFIGPATH=C:/Users/runneradmin/AppData/Roaming/ccache/ccache.conf" >> $GITHUB_ENV
 
-      - name: CMake
+      - name: Set up ccache
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          key: ${{ matrix.runs-on }}
+
+      - name: Configure maplibre-gl-native (MacOS)
         if: runner.os == 'MacOS'
         run: |
-          cmake . -B build -G Ninja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=${{ env.BUILDTYPE }}
+          cmake . -B build \
+            -G Ninja \
+            -DCMAKE_BUILD_TYPE=${{ env.BUILDTYPE }} \
+            -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
 
-      - name: CMake
+      - name: Configure maplibre-gl-native (Linux)
         if: runner.os == 'Linux'
         run: |
-          cmake . -B build -G Ninja -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DCMAKE_BUILD_TYPE=${{ env.BUILDTYPE }} -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10
+          cmake . -B build \
+            -G Ninja \
+            -DCMAKE_BUILD_TYPE=${{ env.BUILDTYPE }} \
+            -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+            -DCMAKE_C_COMPILER=gcc-10 \
+            -DCMAKE_CXX_COMPILER=g++-10
+
+      - name: Configure maplibre-gl-native (Windows)
+        if: runner.os == 'Windows'
+        shell: cmd
```

Comment:
can this be moved back to bash with the other syntax. the ccache plugin we are using notes

```
Notes on Windows support
Note that using Ccache on Windows is still experimental and https://github.com/ccache/ccache/issues/1023. Use Sccache for stable Windows support.
```
https://github.com/hendrikmuhs/ccache-action#notes-on-windows-support